xml_converter/src/xml_converter.cpp

```diff
@@ -308,6 +314,10 @@ int main(int argc, char* argv[]) {
             allow_duplicates = true;
             arg_target = nullptr;
         }
+        else if (!strcmp(argv[i], "--split-guildpoint-by-category")) {
+            split_proto_by_category = true;
```

Comment:
What happens if `--output-split-guildpoint-path` and `--split-guildpoint-by-category` are both specified. Does the program act in a way that a user could reasonably expect? I think it wont, because it will only split by map id as the current logic for splitting by map id does not interact with the splitting by category logic.

Maybe this is the wrong argument to be exposing to the user? Think through some user stories for this holistically because we can still alter how other command line flags for xml_converter are presented to the user if we think there are better paradigms to use than what we are currently using.